src/plugins_3rdparty/primer3/src/Primer3TaskSettings.cpp

```diff
@@ -49,6 +49,18 @@ Primer3TaskSettings::Primer3TaskSettings() {
     primerSettings = p3_create_global_settings_default_version_1();
     seqArgs = create_seq_arg();
     isCircular = false;
+    // UGENE numerates sequence characters in a human-readable mode - the first character has index 1, like this:
+    // Index number:       1234...
+    // Sequence character: ACGT...
+    //
+    // Primer3 numerates characters (by default) in a programming mode - the first character has index 0:
+    // Index number:       0123...
+    // Sequence character: ACGT...
+    //
+    // It causes one character shift is the result (UGENE-7825)
+    // But Primer3 has a special parameter to set a number to start indexing with - FIRST_BASE_INDEX (default value = 0)
+    // By settings this parameter to 1 we make Primer3 index characters from number 1
+    primerSettings->first_base_index = 1;
```

Comment:
Does it mean that UGENE in internal model for primer position in sequence uses math (U2Regions) that are 1 based, but not 0 based?

Reply:
Yes, I'm talking about UI rendering in this case. Previous version of Primer3 also has this value set to 1, I just missed it when was doing upgrading

Reply:
I thought Primer3TaskSettings is a data structure to configure how Primer3 algorithm works (non-ui).
And the default mode for primer3 is 0.
This way we can completely remove the method Primer3TaskSettings::getFirstBaseIndex and all logic around it.

But you do it the opposite way: set it to a non default value and adds logic around this parameter.

Could you please give more details why can't we use 0 as default and completely remove getFirstBaseIndex from our code?

Reply:
I found that we do change this property in our code - so it is not readonly. Resolving.